### ISSUE TITLE
Handle exceptions in ElevationSource worker threads better

### ISSUE THREAD

**Issue by TaylorMutch** [VISTAS-IVES/pyvistas#78]
It is not unlikely that we encounter a case where a thread does not return gracefully. It would be good to have a standard way for handling when a thread throws an unexpected `Exception`.

**Comment by TaylorMutch** [comment]
Example - when acquiring tile meshes, sometimes the `ElevationSource` can't find the tile it's looking for, or it hasn't acquired the tile from the remote resource yet, but we're asking for it here. It may just be that we need to add an `async` to `ElevationSource.create_data_dem` and hold until the event loop returns with the data, or we throw an exception that is handled in the `Thread` that we can then listen for and post a message back to the UI.
```
Exception in thread Thread-17:
Traceback (most recent call last):
  File "C:\Python35-64\lib\threading.py", line 914, in _bootstrap_inner
    self.run()
  File "C:/Users/Taylor/Documents/pyvistas/source\vistas\core\graphics\tile.py", line 134, in run
    data = grids[t].T
KeyError: Tile(x=19, y=44, z=7)
```

**Comment by TaylorMutch** [comment]
Closed #78.
